src/WebJobs.Script.Abstractions/Description/Binding/BindingMetadata.cs

```diff
@@ -81,6 +88,10 @@ public static BindingMetadata Create(JObject raw)
             string bindingDirectionValue = (string)raw["direction"];
             string connection = (string)raw["connection"];
             string bindingType = (string)raw["type"];
+            IDictionary<string, object> properties = raw.TryGetValue("properties", StringComparison.OrdinalIgnoreCase, out JToken value)
+                            ? new Dictionary<string, object>(value.ToObject<IDictionary<string, object>>(), StringComparer.OrdinalIgnoreCase)
+                            : ImmutableDictionary<string, object>.Empty;
```

Comment:
Is `ImmutableDictionary` correct here? It implements `IDictionary` sure, but will it behave as expected? What will happen if someone tries to mutate it? `Properties` is technically writeable after all.

Reply:
That is a good question, Fabio originally made this suggestion, but maybe not realizing Properties has a setter? Maybe in this case just `new Dictionary<string, object>` make sense?

@fabiocav thoughts?

Reply:
> Properties has a setter

It isn't that this has a setter or not, it is that this type is `IDictionary<>`, which is a _mutable_ contract. While `ImmutableDictionary` does implement it, it throws `NotSupportedException` anytime you try to actually use a [mutable API from it](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableDictionary_2.cs#L264). The concern is this could cause very obscure bugs due to the concrete type, and behavior, behind `Properties` changing significantly based on some pre-conditions.

Reply:
Yup, that makes sense! Thanks for the details :) We are writing to the properties dictionary in another PR, but only if the dictionary contains a value, which in that case the type will be Dictionary and not IImmutableDictionary. But I can see where this will cause issues down the line - I'm happy to change this back to Dictionary<string, object>!

Reply:
You might consider making Properties behave the same way FunctionMetadata.Properties does https://github.com/Azure/azure-functions-host/blob/6bfbe194b317b2e20835f9c5ed8f9f6ab8d78765/src/WebJobs.Script.Abstractions/Description/FunctionMetadata.cs#L17. We initialize it in the default ctor, and the property itself has no setter.

Reply:
Ooo, I like that idea!